### PR TITLE
Add Microsoft Entra ID Entity Analytics Integration

### LINKED ISSUE #7037: [Azure Entity Analytics] Releasing AzureAD EA Package
The Azure Entity Analytics package is available internally, but not yet publicly release. We are now at a point where we can ship the integration, but have some loose ends to tie up before doing so:

- Adjust title to 'Microsoft Entra ID' due to AzureAD's rebranding exercise. 
- Adjust description to: 'Collect user entities from Entra ID (formerly AzureAD)` with Elastic Agent. 
- Add dashboards to the package, inline with the dashboards we've shipped with the Okta EA package [[screenshot](https://github.com/elastic/integrations/blob/main/packages/entityanalytics_okta/img/entityanalytics_okta-user-screenshot.png)]

### PR DESCRIPTION
**Pull request by taylor-swanson**
## What does this PR do?

- Initial draft of package (see #5638)
- Rebrand Azure AD to Microsoft Entra ID
- Add pipeline and system tests
- Add docs

Note: Dashboards and support for devices will be added in later PRs.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

```
cd package/entityanalytics_entra_id
elastic-package test
```

## Related issues

- Closes #7037

### CODE REVIEW

File: packages/entityanalytics_entra_id/changelog.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# newer versions go on top
+- version: "0.1.0"
+  changes:
+    - description: Initial release of package.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/7392
+- version: "1.0.0-next"
```
Comment (efd6) [inline packages/entityanalytics_entra_id/changelog.yml:7]: Given that this was not released, maybe roll this version into the version above?

File: packages/entityanalytics_entra_id/data_stream/entity/agent/stream/entity-analytics.yml.hbs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+provider: azure-ad
+sync_interval: {{sync_interval}}
+update_interval: {{update_interval}}
+client_id: {{client_id}}
+tenant_id: {{tenant_id}}
+secret: {{secret}}
+login_endpoint: {{login_endpoint}}
+api_endpoint: {{api_endpoint}}
+login_scopes:
+{{#each login_scopes as |login_scope i|}}
```
Comment (efd6) [inline packages/entityanalytics_entra_id/data_stream/entity/agent/stream/entity-analytics.yml.hbs:10]:
```suggestion
{{#each login_scopes as |login_scope|}}
```

**Comment by taylor-swanson** [comment]
@efd6, any other thoughts?

The only other thing I was questing was the name of the data stream. I know we are talking about using the reroute processor to put user and device docs into their respective data streams, but I imagine we still need an intermediate data stream (even if it's not used in the end) to host the pipeline? I'm not sure how you'd structure that in the integration, but at least for now I'll put it into a generically named data stream (`entity`).

**Review by efd6 — APPROVED** [review]
I think entity is a perfectly reasonable name for the generic case.

LGTM

**Pull request merged**